apps/cave/components/Transparency/Charts/useFetchData.ts

```diff
@@ -0,0 +1,31 @@
+import { NEXT_PUBLIC_CHART_ENDPOINT } from 'lib/env.conf'
+import { useQuery, UseQueryResult } from 'react-query'
+import { fetchData } from './fetchData'
+
+export const useFetchData = <T>(
+  route: string,
+  url: string = NEXT_PUBLIC_CHART_ENDPOINT,
+  method?: string,
+  bodyData?: object,
+): UseQueryResult<T, unknown> => {
+  return useQuery(
+    [
+      JSON.stringify({
+        route,
+        url,
+        method,
+        bodyData,
+      }),
+    ],
+    () => fetchData<T>(route, url, method, bodyData),
+    {
+      retry: 3,
+      refetchInterval: 15000,
```

Comment:
want to do a refetch every 10-15 minutes?

realistically we don't need to refetch too often, iirc the data gets updated every 4 hours

Reply:
👍